src/main/java/org/openpnp/spi/base/AbstractPnpJobProcessor.java

```diff
@@ -36,7 +49,12 @@ public static void discard(Nozzle nozzle) throws Exception {
                 Configuration.get().getMachine().getDiscardLocation());
         // discard the part
         nozzle.place();
+		// repeat the air puff to clean filters over discard location
```

Comment:
This is too machine specific and should be handled at the driver level. It will cause solenoids to trigger repeatedly and will confuse users who do not have this filter.